src/form/df.jl

```diff
@@ -0,0 +1,163 @@
+# this file contains (balanced) convexified DistFlow formulation, in W space
+export
+    SOCDFPowerModel, SOCDFForm
+""
+abstract type AbstractDFForm <: AbstractPowerFormulation end
+
+""
+abstract type SOCDFForm <: AbstractDFForm end
+
+""
+const SOCDFPowerModel = GenericPowerModel{SOCDFForm}
+
+"default SOC constructor"
+SOCDFPowerModel(data::Dict{String,Any}; kwargs...) = GenericPowerModel(data, SOCDFForm; kwargs...)
+
+""
+function variable_branch_current(pm::GenericPowerModel{T}, n::Int=pm.cnw; kwargs...) where T <: AbstractDFForm
+    variable_branch_series_current_magnitude_sqr(pm, n; kwargs...)
+end
+
+""
+function variable_branch_flow(pm::GenericPowerModel{T}, n::Int=pm.cnw; kwargs...) where T <: AbstractDFForm
+    variable_active_branch_flow(pm, n; kwargs...)
+    variable_reactive_branch_flow(pm, n; kwargs...)
+end
+
+""
+function variable_voltage(pm::GenericPowerModel{T}, n::Int=pm.cnw; kwargs...) where T <: AbstractDFForm
+    variable_voltage_magnitude_sqr(pm, n; kwargs...)
+end
+
+"do nothing, this model does not have complex voltage constraints"
+function constraint_voltage(pm::GenericPowerModel{T}, n::Int) where T <: AbstractDFForm
```

Comment:
This implementation can be provided without any constraint on `T` and can be placed in `core/constraint.jl`

Reply:
ok